C++ API parity: Tanhshrink

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* #27538 C++ API parity: Threshold
* **#27537 C++ API parity: Tanhshrink**

Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

pbelevich committed:
C++ API parity: Tanhshrink

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"




[ghstack-poisoned]

yf225 reviewed (approved):
Looks awesome, thanks a lot @pbelevich!

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Tanhshrink"


Differential Revision: [D17835409](https://our.internmc.facebook.com/intern/diff/D17835409)

[ghstack-poisoned]
